<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RNSQTClass</class>
 <widget class="QMainWindow" name="RNSQTClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1089</width>
    <height>674</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Решение задач цифровой обработки сигналов в системе остаточных классов</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="gbModules">
        <property name="title">
         <string>Модули СОК</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="1">
          <widget class="QSpinBox" name="sbRNSm2">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
           <property name="value">
            <number>79</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="sbRNSm3">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
           <property name="value">
            <number>97</number>
           </property>
          </widget>
         </item>
         <item row="1" column="3" colspan="2">
          <widget class="QPushButton" name="btnCheckModuli">
           <property name="text">
            <string>Проверить модули</string>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QSpinBox" name="sbRNSm6">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QSpinBox" name="sbRNSm4">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
           <property name="value">
            <number>109</number>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QSpinBox" name="sbRNSm5">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
           <property name="value">
            <number>353</number>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QSpinBox" name="sbRNSm7">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="7">
          <widget class="QSpinBox" name="sbRNSm8">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="8">
          <widget class="QSpinBox" name="sbRNSm9">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="9">
          <widget class="QSpinBox" name="sbRNSm10">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QSpinBox" name="sbRNSm1">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
           <property name="value">
            <number>73</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QSpinBox" name="sbModuliCount">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="singleStep">
            <number>1</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QLabel" name="lblModuliCount">
           <property name="text">
            <string>Количество модулей:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="5" colspan="2">
          <widget class="QLabel" name="lblMaxValue">
           <property name="text">
            <string>Максимальное число СОК:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="7" colspan="2">
          <widget class="QLineEdit" name="edMaxValue">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gbFirst">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>Первый операнд</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="3">
          <widget class="QSpinBox" name="sbo1m4">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QSpinBox" name="sbo1m6">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QSpinBox" name="sbo1m1">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="sbo1m2">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="sbo1m3">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QSpinBox" name="sbo1m5">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QSpinBox" name="sbo1m7">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="7">
          <widget class="QSpinBox" name="sbo1m8">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="8">
          <widget class="QSpinBox" name="sbo1m9">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="9">
          <widget class="QSpinBox" name="sbo1m10">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="lbl1Result">
           <property name="text">
            <string>В десятичной:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3" colspan="2">
          <widget class="QLineEdit" name="ed1Result"/>
         </item>
         <item row="1" column="6">
          <widget class="QLabel" name="lblMultiplier1">
           <property name="text">
            <string>Порядок:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="7">
          <widget class="QSpinBox" name="sbo1Factor">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>-9</number>
           </property>
           <property name="maximum">
            <number>9</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gbSecond">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>Второй операнд</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="5">
          <widget class="QSpinBox" name="sbo2m6">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QSpinBox" name="sbo2m4">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="sbo2m3">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QSpinBox" name="sbo2m1">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="sbo2m2">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QSpinBox" name="sbo2m5">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QSpinBox" name="sbo2m7">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="7">
          <widget class="QSpinBox" name="sbo2m8">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="8">
          <widget class="QSpinBox" name="sbo2m9">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="9">
          <widget class="QSpinBox" name="sbo2m10">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="7">
          <widget class="QSpinBox" name="sbo2Factor">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>-9</number>
           </property>
           <property name="maximum">
            <number>9</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="lbl2Result">
           <property name="text">
            <string>В десятичной:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="6">
          <widget class="QLabel" name="lblMultiplier2">
           <property name="text">
            <string>Порядок:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3" colspan="2">
          <widget class="QLineEdit" name="ed2Result"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gbOperation">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>Операция</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QRadioButton" name="rbOpSum">
           <property name="text">
            <string>Сложение</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QRadioButton" name="rbOpMul">
           <property name="text">
            <string>Умножение</string>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QPushButton" name="btnOperate1CPU">
           <property name="text">
            <string>Выполнить в 1 CPU</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QRadioButton" name="rbOpDiv">
           <property name="text">
            <string>Целочисленное деление</string>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QPushButton" name="btnOperateGPU">
           <property name="text">
            <string>Выполнить в GPU</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QRadioButton" name="rbOpSub">
           <property name="text">
            <string>Вычитание</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gbTotalResult">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="title">
         <string>Результат</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="5">
          <widget class="QSpinBox" name="sbRm6">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QSpinBox" name="sbRm4">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="sbRm3">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QSpinBox" name="sbRm1">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="sbRm2">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QSpinBox" name="sbRm5">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QSpinBox" name="sbRm7">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="7">
          <widget class="QSpinBox" name="sbRm8">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="8">
          <widget class="QSpinBox" name="sbRm9">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="9">
          <widget class="QSpinBox" name="sbRm10">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="7">
          <widget class="QSpinBox" name="sboResultFactor">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="minimum">
            <number>-9</number>
           </property>
           <property name="maximum">
            <number>9</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="lbl2Result_4">
           <property name="text">
            <string>В десятичной:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="6">
          <widget class="QLabel" name="lblMultiplier2_4">
           <property name="text">
            <string>Порядок:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3" colspan="2">
          <widget class="QLineEdit" name="edTotalResult">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="gbOut">
        <property name="title">
         <string>Консоль</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="2" column="0">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="3">
            <widget class="QPushButton" name="btnAbout">
             <property name="text">
              <string>О программе</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="4">
            <widget class="QPlainTextEdit" name="pteOut">
             <property name="palette">
              <palette>
               <active>
                <colorrole role="Text">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>0</red>
                   <green>255</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Base">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>0</red>
                   <green>0</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
               </active>
               <inactive>
                <colorrole role="Text">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>0</red>
                   <green>255</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Base">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>0</red>
                   <green>0</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
               </inactive>
               <disabled>
                <colorrole role="Text">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>120</red>
                   <green>120</green>
                   <blue>120</blue>
                  </color>
                 </brush>
                </colorrole>
                <colorrole role="Base">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>240</red>
                   <green>240</green>
                   <blue>240</blue>
                  </color>
                 </brush>
                </colorrole>
               </disabled>
              </palette>
             </property>
             <property name="font">
              <font>
               <family>Courier New</family>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOn</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="btnLoad">
             <property name="text">
              <string>Загрузить параметры</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="btnSave">
             <property name="text">
              <string>Сохранить параметры</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <slots>
  <slot>slotButton()</slot>
  <slot>slotModuliCountChanged(int)</slot>
  <slot>slotChangeModuli()</slot>
 </slots>
</ui>
